working/1426-extension-types/feature-specification-implements-rep.md

```diff
@@ -0,0 +1,268 @@
+`R` and assume that the `implements` clause of _DV_ includes the non-inline
+types `R1 .. Rk`. *We then have `R <: Rj` for each `j`, because anything
+else is an error.*
+
+Assume that `m` is a member which not declared by _DV_, and none of _DV_'s
+inline superinterfaces have a member named `m`, but one or more of the
+interfaces of `R1 .. Rk` has a member named `m`. A compile-time error
+occurs if there exist `j1` and `j2` in `1 .. k` and a member name `m` such
+that `m` does not have a combined member signature for `R1
+.. Rk`. Otherwise the member signature of `m` is that combined member
+signature.
+
+Invocations of members declared by _DV_ or declared by an inline
+superinterface of _DV_ and not declared by any of `Rj`, `j` in `1..k` are
+unaffected by the fact that _DV_ implements `R1 .. Rk`.
+
+*This could be an invocation of a member declared by _DV_, or by any of its
+non-inline superinterfaces, or both, but the rules are unchanged.*
+
+Let `m` be a member name which is not declared by _DV_. Assume that the
+interface of `Rj` has a member named `m`. A compile-time error occurs if an
+inline superinterface of _DV_ also declares a member named `m`.
+
+Let `m` be a member name which is not declared by _DV_ and not declared by
+an inline superinterface of _DV_. Assume that the interface of `Rj` has a
+member named `m` with signature `s` *(this is the combined member signature
+that may depend on other types in `R1 .. Rk`)*. An invocation of `m` on a
+receiver of type `V` (or `V<T1 .. Ts>` if _DV_ is generic) is then treated
+as the same invocation, but with signature `s`.
+
+*In other words: Conflicts among superinterfaces are treated the same,
+whether it is an inline or a non-inline superinterface. In both cases, _DV_
+can resolve the conflict by redeclaring the given member. No override check
+is applied, any signature with the given name will resolve the conflict. If
+there is no conflict then _DV_ will "forward to" the members of `R0`.*
+
+*An implementation may choose to implicitly induce forwarding members into
+_DV_ in order to enable invocation of members of `R0`. However, such
+forwarding members must preserve the semantics of a direct invocation. In
+particular, if an invocation omits some optional parameters then the
+invocation of a member of `R0` must use the default value for that
+parameter of the actually invoked instance method, not a statically known
+value.*
+
+```dart
+class C {
+  int m([int i = 0]) => i;
+}
+
+class D extends C {
+  int m([int i = 42, j = -1]) => i;
+}
+
+inline class V implements C {
+  final C it;
+  V(this.it);
+}
+
+void main() {
+  V v = V(D());
+  // v.m(3, 4); // Compile-time error: `m` signature is from `C`.
+  v.m(); // Returns 42, not 0.
+}
+```
+
+It is an implementation specific behavior whether a closurization
+*(also known as a tear-off)* of an inline class instance member which is
+obtained from the interface of `R0` is a tear-off of the member of the
```

Comment:
Yes, changed that.

It is not guaranteed that there exists a combined interface for `R1 .. Rk`, but each member which isn't redeclared has a combined member signature (otherwise we have a compile-time error). I'm currently referring to "a member of `R1 .. Rk`" and relying on earlier text where the member signature must be achieved by combination.